modules/dnn/src/layers/reshape_layer.cpp

```diff
@@ -169,14 +169,13 @@ class ReshapeLayerImpl : public ReshapeLayer
     {
         for (size_t i = 0; i < inputs.size(); i++)
         {
-            Mat srcBlob = *inputs[i];
-            MatShape inputShape = shape(srcBlob), outShape = shape(outputs[i]);
+            Mat& srcBlob = *inputs[i];
+            Mat& dstBlob = outputs[i];
+            MatShape inputShape = shape(srcBlob);
 
             if (performReordering)
             {
-                Mat reordered_blob(inputShape, srcBlob.type());
-
-                float *dstData = reordered_blob.ptr<float>();
+                float *dstData = dstBlob.ptr<float>();
```

Comment:
Chosen the second way of implementation. Thanks!
Also, modified reordering conditions.